clang/include/clang/CIR/Interfaces/ASTAttrInterfaces.td

```diff
@@ -64,10 +64,42 @@ let cppNamespace = "::cir" in {
             MangleCtx->mangleDynamicInitializer($_attr.getAst(), Out);
           }]
       >,
+      InterfaceMethod<"", "void", "mangleStaticGuardVariable", (ins "llvm::raw_ostream&":$Out), [{}],
+        /*defaultImplementation=*/ [{
+          std::unique_ptr<clang::MangleContext> mangleCtx(
+            $_attr.getAst()->getASTContext().createMangleContext());
+            mangleCtx->mangleDynamicInitializer($_attr.getAst(), Out);
+          }]
+      >,
       InterfaceMethod<"", "clang::VarDecl::TLSKind", "getTLSKind", (ins), [{}],
         /*defaultImplementation=*/ [{
           return $_attr.getAst()->getTLSKind();
         }]
+      >,
+      InterfaceMethod<"", "bool", "isInline", (ins), [{}],
+        /*defaultImplementation=*/ [{
+          return $_attr.getAst()->isInline();
+        }]
+      >,
+      InterfaceMethod<"", "clang::TemplateSpecializationKind", "getTemplateSpecializationKind", (ins), [{}],
+        /*defaultImplementation=*/ [{
+          return $_attr.getAst()->getTemplateSpecializationKind();
+        }]
+      >,
+      InterfaceMethod<"", "bool", "isLocalVarDecl", (ins), [{}],
+        /*defaultImplementation=*/ [{
+          return $_attr.getAst()->isLocalVarDecl();
+        }]
+      >,
+      InterfaceMethod<"", "clang::SourceLocation", "getLocation", (ins), [{}],
+        /*defaultImplementation=*/ [{
+          return $_attr.getAst()->getLocation();
+        }]
+      >,
+      InterfaceMethod<"", "const clang::VarDecl *", "getRawDecl", (ins), [{}],
```

Comment:
@bcardosolopes Thoughts on this? This wasn't scaling to the real world. e.g. functions that take a `VarDecl` as an argument.

Reply:
After you latest update you can probably ditch this altogether